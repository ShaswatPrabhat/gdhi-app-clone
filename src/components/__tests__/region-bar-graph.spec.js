import { mount } from "@vue/test-utils";
import { describe, beforeEach, it, expect, vi, vitest } from "vitest";
import RegionBarGraph from "../graphs/region-bar-graph/region-bar-graph.vue";
import Chart from "chart.js/auto";
import { i18n } from "../../plugins/i18n";

describe("Region Bar Graph Liner", () => {
  let wrapper;
  vi.mock("chart.js/auto");
  const defaultYearCountriesScore = new Map();
  defaultYearCountriesScore.set("Nigeria", {
    categories: [
      {
        id: 1,
        name: "Leadership & Governance",
        overallScore: 3.5,
        phase: 4,
        indicators: null,
      },
      {
        id: 2,
        name: "Strategy & Investment",
        overallScore: 2.5,
        phase: 3,
        indicators: null,
      },
      {
        id: 3,
        name: "Legislation, Policy, & Compliance",
        overallScore: 2.75,
        phase: 3,
        indicators: null,
      },
      {
        id: 4,
        name: "Workforce",
        overallScore: 1.5,
        phase: 2,
        indicators: null,
      },
      {
        id: 5,
        name: "Standards & Interoperability",
        overallScore: 1,
        phase: 1,
        indicators: null,
      },
      {
        id: 6,
        name: "Infrastructure",
        overallScore: 1.5,
        phase: 2,
        indicators: null,
      },
      {
        id: 7,
        name: "Services & Applications",
        overallScore: 1.6666666666666667,
        phase: 2,
        indicators: null,
      },
    ],
    countryPhase: 3,
  });
  const selectedYearCountriesScore = new Map();
  selectedYearCountriesScore.set("Malawi", {
    categories: [
      {
        id: 1,
        name: "Leadership & Governance",
        overallScore: 2,
        phase: 2,
        indicators: null,
      },
      {
        id: 2,
        name: "Strategy & Investment",
        overallScore: 3.5,
        phase: 4,
        indicators: null,
      },
      {
        id: 3,
        name: "Legislation, Policy, & Compliance",
        overallScore: 4,
        phase: 4,
        indicators: null,
      },
      {
        id: 4,
        name: "Workforce",
        overallScore: 3.5,
        phase: 4,
        indicators: null,
      },
      {
        id: 5,
        name: "Standards & Interoperability",
        overallScore: 4,
        phase: 4,
        indicators: null,
      },
      {
        id: 6,
        name: "Infrastructure",
        overallScore: 4.5,
        phase: 5,
        indicators: null,
      },
      {
        id: 7,
        name: "Services & Applications",
        overallScore: 4,
        phase: 4,
        indicators: null,
      },
    ],
    countryPhase: 4,
  });
  beforeEach(() => {
    vitest.useFakeTimers();
    wrapper = mount(RegionBarGraph, {
      propsData: {
        selectedYearData: selectedYearCountriesScore,
        defaultYearData: defaultYearCountriesScore,
        categoryFilter: 0,
        labels: [],
      },
      i18n,
    });
  });
  it("should render Graph when mounted and create a new Chart instance", () => {
    vitest.advanceTimersByTime(100);
    expect(wrapper.vm.$el).toMatchSnapshot();
    expect(Chart.mock.calls.length).toEqual(1);
    expect(Chart.mock.calls[0][1]).toMatchSnapshot();
  });
});
