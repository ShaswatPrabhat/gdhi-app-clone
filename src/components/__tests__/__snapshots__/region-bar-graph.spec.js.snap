// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Region Bar Graph Liner > should render Graph when mounted and create a new Chart instance 1`] = `
<div
  class="graph"
  data-v-6edbe18b=""
>
  <div
    class="graph-container"
    data-v-6edbe18b=""
    style="--width: 1500px; --height: 550px;"
  >
    <canvas
      data-v-6edbe18b=""
      id="bar-graph"
    />
  </div>
</div>
`;

exports[`Region Bar Graph Liner > should render Graph when mounted and create a new Chart instance 2`] = `
{
  "data": {
    "datasets": [
      {
        "backgroundColor": "#CED4DA",
        "barPercentage": 1,
        "barThickness": 60,
        "categoryPercentage": 0.5,
        "data": [],
        "label": "Selected year",
      },
      {
        "backgroundColor": [Function],
        "barPercentage": 1,
        "barThickness": 60,
        "categoryPercentage": 0.5,
        "data": [],
        "label": "Default year",
      },
    ],
    "labels": [],
  },
  "options": {
    "animation": false,
    "clip": false,
    "layout": {
      "padding": {
        "bottom": 0,
        "left": 60,
        "right": 80,
        "top": 60,
      },
    },
    "maintainAspectRatio": false,
    "plugins": {
      "legend": {
        "display": false,
      },
      "tooltip": {
        "rtl": false,
      },
    },
    "responsive": true,
    "scales": {
      "x": {
        "border": {
          "color": "#ced4da",
          "dash": [
            2,
            4,
          ],
          "width": 1,
        },
        "grid": {
          "display": true,
          "drawTicks": true,
          "lineWidth": 0,
          "tickColor": "white",
          "tickLength": 16,
          "tickWidth": 1.5,
        },
        "reverse": false,
        "ticks": {
          "backdropPadding": {
            "bottom": 100,
            "left": 100,
            "right": 100,
            "top": 100,
          },
          "callback": [Function],
          "color": "#0A0A0A",
          "font": {
            "lineHeight": "16.94px",
            "size": 16,
            "weight": 400,
          },
        },
      },
      "y": {
        "border": {
          "color": "#ced4da",
          "dash": [
            2,
            2,
          ],
          "width": 0,
        },
        "grid": {
          "drawTicks": true,
          "lineWidth": 1,
          "tickColor": "white",
          "tickLength": 16,
        },
        "max": 5,
        "min": 0,
        "position": "left",
        "ticks": {
          "callback": [Function],
          "color": "#0A0A0A",
          "font": {
            "size": 16,
            "weight": 400,
          },
        },
      },
    },
  },
  "type": "bar",
}
`;
